### PR TITLE
Stop using deprecated Envoy runtimes flags

### PR DESCRIPTION
**Pull request by howardjohn**
Fixes https://github.com/istio/istio/issues/45462

I think this is the best we can do without Envoy reverting

### CODE REVIEW

File: pkg/bootstrap/config.go [hunk 1]
```diff
@@ -318,6 +318,17 @@ func extractRuntimeFlags(cfg *model.NodeMetaProxyConfig) map[string]string {
...
+		// Convert strings to bools for backwards compat.
+		switch v {
+		case "false":
+			runtimeFlags[k] = false
+		case "true":
+			runtimeFlags[k] = true
+		default:
+			runtimeFlags[k] = v
+		}
 		runtimeFlags[k] = v
```
Comment (kyessenov) [inline pkg/bootstrap/config.go:332]: delete this line?

Comment (howardjohn) [reply pkg/bootstrap/config.go:332]: thanks. probably a sign we need a test

**Review by kyessenov — APPROVED** [review]
Envoy only uses number and bool runtime values, so it's very unlikely literal "true" is used anywhere as a string value.

File: pkg/bootstrap/config.go (hunk 1 above)
Comment (kyessenov) [reply pkg/bootstrap/config.go:332]: yes, absolutely.

**Comment by ramaraochavali** [comment]
LGTM

**Pull request merged**